docs/builtin-classes.md

```diff
@@ -0,0 +1,396 @@
+)
+
+fun main() {
+    val data = Data(0x1CAFE2FEED0BABE0)
+    println(Json.encodeToString(data))
+}
+``` 
+
+> You can get the full code [here](../guide/example/example-builtin-04.kt).
+
+This JSON gets parsed natively by JavaScript without loss of precision.
+
+```text
+{"signature":"2067120338512882656"}
+```                                         
+
+> The section on [Specifying serializers for a file](serializers.md#specifying-serializers-for-a-file) explains how a 
+> serializer like `LongAsStringSerializer` can be specified for all properties in a file.
+
+<!--- TEST -->
+
+### Enum classes
+
+All enum classes are serializable out of the box without having to mark them `@Serializable`,
+as the following example shows.
+
+```kotlin          
+// @Serializable annotation is not need for a enum classes
+enum class Status { SUPPORTED }
+        
+@Serializable
+class Project(val name: String, val status: Status) 
+
+fun main() {
+    val data = Project("kotlinx.serialization", Status.SUPPORTED)
+    println(Json.encodeToString(data))
+}
+```                                        
+
+> You can get the full code [here](../guide/example/example-builtin-05.kt).
+
+In JSON enum gets encoded as a string.
+
+```text
+{"name":"kotlinx.serialization","status":"SUPPORTED"}
+```   
+
+<!--- TEST -->
+
+### Serial names of enum entries
+
+Serial names of enum entries can be customized with [SerialName] annotation just like 
+it was shown for properties in the [Serial field names](basic-serialization.md#serial-field-names) section.
+However, in this case the whole enum class must be marked with the [`@Serializable`][Serializable] annotation.
+
+```kotlin
+@Serializable // required because of @SerialName
+enum class Status { @SerialName("maintained") SUPPORTED }
+        
+@Serializable
+class Project(val name: String, val status: Status) 
+
+fun main() {
+    val data = Project("kotlinx.serialization", Status.SUPPORTED)
+    println(Json.encodeToString(data))
+}
+```                                        
+
+> You can get the full code [here](../guide/example/example-builtin-06.kt).
+
+We see that the specified serial name is now used in the resulting JSON.
+
+```text
+{"name":"kotlinx.serialization","status":"maintained"}
+```   
+
+<!--- TEST -->
+
+## Composites
+
+A number of composite types from the standard library are supported by Kotlin serialization.
+
+### Pair and triple
+
+Simple data classes [Pair] and [Triple] from the Kotlin standard library are serializable.
+
+```kotlin
+@Serializable
+class Project(val name: String)
+
+fun main() {
+    val pair = 1 to Project("kotlinx.serialization")
+    println(Json.encodeToString(pair))
+}  
+```                                
+
+> You can get the full code [here](../guide/example/example-builtin-07.kt).
+
+```text
+{"first":1,"second":{"name":"kotlinx.serialization"}}
+```
+
+<!--- TEST -->
+ 
+> Not all classes from the standard library are serializable. In particular ranges and [Regex] class
```

Comment:
```suggestion
> Not all classes from the standard library are serializable. In particular, ranges and [Regex] class
```

Reply:
```suggestion
> Not all classes from the Kotlin standard library are serializable. In particular ranges and [Regex] class
```

It may be helpful to add the qualifier that we are talking about the Kotlin standard library (not the Java/JS standard library)